atomate/atat/fireworks/core.py

```diff
@@ -0,0 +1,193 @@
+from fireworks import ScriptTask, FileWriteTask
+from monty.os.path import which
+from atomate.utils.utils import get_logger
+from atomate.atat.firetasks.parse_outputs import McsqsToDbTask
+from atomate.atat.utilities import preprocess_disordered_structure
+from fractions import Fraction
+from string import Template
+
+__author__ = 'Matthew Horton'
+__credits__ = 'Josua Vieten'
+__email__ = 'mkhorton@lbl.gov'
+
+logger = get_logger(__name__)
+module_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)))
+
+class McsqsFW(Firework):
+
+    def __init__(self, disordered_struct,
+                 name="mcsqs",
+                 size=None,
+                 clusters=None,
+                 user_input_settings=None,
+                 max_denominator=8,
+                 walltime=60,
+                 ncores=None,
+                 db_file=None,
+                 **kwargs):
+        """
+        Find a best SQS approximation for a given disordered
+        structure using the mcsqs code from the ATAT library.
+
+        One unusual feature of this Firework: walltime must
+        be specified. This is because there are often cases
+        where mcsqs will never terminate (it will keep trying
+        structures at random); however, whatever structure
+        it finds in this time (the 'bestsqs') is often
+        still useful.
+
+        :param disordered_struct: disordered structure
+        :param name: name of the Firework
+        :param size: ratio of atoms in the output supercell
+        to number of atoms in input cell, if None will
+        choose the smallest possible supercell
+        :param clusters: dict of cluster sizes, e.g.
+        {2: 1.5, 3: 1.0} to search for pairs of atoms
+        within a radius of 1.5 Å and triplets of atoms
+        within a radius of 1.0 Å, if None will try to
+        pick sensible defaults, the value of mcsqs is
+        that this should converge rapidly with number of
+        clusters (aim for around a dozen)
+        :param user_input_settings: dict of additional keyword
+        value pairs to pass to mcsqs, such as Monte Carlo
+        temperature, no validation is performed on this dict
+        :param: max_denominator (int):
+        Will round occupancies to the nearest rational number,
+        with the maximum denominator as specified. Use None not
+        to do perform rounding, though not mcsqs might not like this!
+        :param: walltime (int): Time to run mcsqs for in minutes,
+        1 minute will be deducted from this to run the database
+        tasks.
+        :param: ncores (int): number of instances of mcsqs
+        to run (mcsqs is not parallelized, these are run
+        independently of each other), by default will try
+        to detect automatically
+        :param: db_file:
+        :param kwargs: Other kwargs that are passed to Firework.__init__.
+        """
+
+        if walltime < 3:
+            raise ValueError("Please increase walltime.")
```

Comment:
more helpful message

"Walltime must be at least 3"
or
"Walltime must be at least 3, got {}".format(walltime_mins)